controllers/rest/default_controllers.go

```diff
@@ -279,7 +294,56 @@ func DefaultListController(api Container, commandDispatcher model.CommandDispatc
 				Page:  page,
 				Items: contentEntities,
 			}
-			return ctxt.JSON(http.StatusOK, resp)
+
+			if responseType == "application/json" {
+				return ctxt.JSON(http.StatusOK, resp)
+			} else if responseType == "text/csv" {
+
+				// generate csv
+				w := ctxt.Response().Writer
+				w.Header().Set("Content-Type", responseType)
```

Comment:
also set the Content-DIsposition which will make it so that the user is prompted to download. For the file name you can use the lowercased name of the entity repository `entityRepository.Name()` plus the current date and time  for a bit of info on the header see - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition